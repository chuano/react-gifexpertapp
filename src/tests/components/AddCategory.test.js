import '@testing-library/jest-dom'
import React from 'react';
import { shallow } from 'enzyme' 
import { AddCategory } from '../../components/AddCategory'

describe('AddCategory test', () => {
  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('debe cambiar la caja de texto', () => {
    const value = 'Hola Mundo'
    wrapper.find("input").simulate("change", { target: { value } })

    expect(wrapper.find("input").prop('value')).toBe(value);
  })
  
  test('no debe postear on submit al tener menos de dos caracteres', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {}})

    expect(setCategories).not.toHaveBeenCalled()
  })

  test('debe llamar setCategories y limpiear input', () => {
    const value = 'Hola Mundo'
    wrapper.find('input').simulate('change', { target: { value } })
    wrapper.find('form').simulate('submit', { preventDefault() {} })

    expect(setCategories).toHaveBeenCalledTimes(1)
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))
    expect(wrapper.find('input').prop('value')).toBe('')
  })
})
