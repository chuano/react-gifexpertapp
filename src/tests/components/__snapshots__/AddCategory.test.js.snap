// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory test debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
